components/zcash_address/src/kind/unified.rs

```diff
@@ -208,37 +218,67 @@ impl TryFrom<(&str, &[u8])> for Address {
     type Error = ParseError;
 
     fn try_from((hrp, buf): (&str, &[u8])) -> Result<Self, Self::Error> {
-        let encoded = f4jumble::f4jumble_inv(buf).ok_or(ParseError::InvalidEncoding)?;
+        fn read_receiver(mut cursor: &mut std::io::Cursor<&[u8]>) -> Result<Receiver, ParseError> {
+            let typecode = CompactSize::read(&mut cursor)
+                .map_err(|e| {
+                    ParseError::InvalidEncoding(format!(
+                        "Failed to deserialize CompactSize-encoded typecode {}",
+                        e
+                    ))
+                })
+                .and_then(|v| u32::try_from(v).map_err(|_| ParseError::InvalidTypecodeValue(v)))?;
```

Comment:
This error-map is unnecessary, as `CompactSize::read` enforces the max size limit, so `v` is guaranteed to fit in a `u32`:
```suggestion
                .and_then(|v| u32::try_from(v).expect("CompactSize::read enforces MAX_SIZE limit"))?;
```